src/scala/com/github/johnynek/bazel_deps/S3URLHandler.scala

```diff
@@ -0,0 +1,400 @@
+package com.github.johnynek.bazel_deps
+
+import java.io.{File, FileInputStream, InputStream}
+import java.net.{InetAddress, URI, URL}
+import java.util.Properties
+import java.util.concurrent.ConcurrentHashMap
+import com.amazonaws.ClientConfiguration
+import com.amazonaws.SDKGlobalConfiguration.{ACCESS_KEY_ENV_VAR, ACCESS_KEY_SYSTEM_PROPERTY, SECRET_KEY_ENV_VAR, SECRET_KEY_SYSTEM_PROPERTY}
+import com.amazonaws.auth._
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.s3.model._
+import com.amazonaws.services.s3.{AmazonS3, AmazonS3Client, AmazonS3URI}
+import com.amazonaws.services.securitytoken.{AWSSecurityTokenService, AWSSecurityTokenServiceClient}
+import com.amazonaws.services.securitytoken.model.{AssumeRoleRequest, AssumeRoleResult}
+import org.apache.ivy.util.url.URLHandler
+import org.apache.ivy.util.{CopyProgressEvent, CopyProgressListener, Message}
```

Comment:
ivy dependencies feel like a large beast to be bringing in, what are they giving us here?

Reply:
I'm not sure, but this solution didn't consider coursier first class, so I'm assuming this is something we can get rid of. I couldn't work on this for quite some time, but now I'm back and I'll probably look into this soon.

Reply:
That would be great, seems like a very useful addition to have